packages/greedy_infomax/src/nupic/research/frameworks/greedy_infomax/models/block_model.py

```diff
@@ -0,0 +1,100 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+#
+# This work was based on the original Greedy InfoMax codebase from Sindy Lowe:
+# https://github.com/loeweX/Greedy_InfoMax
+# The Greedy InfoMax paper can be found here:
+# https://arxiv.org/abs/1905.11786
+# ----------------------------------------------------------------------
+
+
+import torch.nn as nn
+import torch.nn.functional as F
+
+from nupic.research.frameworks.greedy_infomax.models.bilinear_info import BilinearInfo
+from nupic.research.frameworks.greedy_infomax.models.utility_layers import (
+    EmitEncoding,
+    _PatchifyInputs,
+)
+
+
+class BlockModel(nn.Module):
+    """
+    A model that supports Greedy InfoMax experiments. This has been deprecated in
+    favor of the more recent GreedyInfoMaxModel which is more general. This module
+    contains a list of modules which are sequentially iterated through for the
+    forward/encode pass. The distinction between supervised training and unsupervised
+    training is implemented by switching which modules are used for the forward pass
+    versus the encode pass. In the forward pass, the BilinearInfoModules are active
+    and the EmitEncoding module is inactive. In the encode pass, the BilinearInfo
+    modules are inactive and the EmitEncoding modules are active.
+    """
+    def __init__(self, modules, **kwargs):
+        super(BlockModel, self).__init__()
+        self.module_list = nn.ModuleList(modules)
```

Comment:
It would be super helpful to include the params for the `__init__` method in the docstring above, just as you did for `FlattenClassifier`.

Reply:
Done